File: ios/ScreenShareAccPackKit/OTScreenShareAccPackKit/AnnotationImportedViewController.m
//
//  ViewController.m
//  ScreenShareSample
//
//  Created by Xi Huang on 4/26/16.
//  Copyright © 2016 Lucas Huang. All rights reserved.
//

#import "AnnotationImportedViewController.h"
#import <OTScreenShareKit/OTScreenShareKit.h>

@interface AnnotationImportedViewController ()
@property (nonatomic) OTAnnotationScrollView *screenShareView;
@end

@implementation AnnotationImportedViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // screen share view
    UIImage *image = [UIImage imageNamed:@"mvc"];
    UIImageView *imageView = [[UIImageView alloc] initWithImage:image];
    imageView.frame = CGRectMake(0, 0, image.size.width, image.size.height);
    
    self.screenShareView = [[OTAnnotationScrollView alloc] initWithFrame:CGRectMake(0,
                                                                                    0,
                                                                                    CGRectGetWidth([UIScreen mainScreen].bounds),
                                                                                    CGRectGetHeight([UIScreen mainScreen].bounds) - 44)];
    [self.screenShareView addContentView:imageView];
    
    [self.screenShareView initializeToolbarView];
    CGFloat height = self.screenShareView.toolbarView.bounds.size.height;
    self.screenShareView.toolbarView.frame = CGRectMake(0, CGRectGetHeight([UIScreen mainScreen].bounds) - height, self.screenShareView.toolbarView.bounds.size.width, height);
    
    [self.view addSubview:self.screenShareView];
    [self.view addSubview:self.screenShareView.toolbarView];
}

@end

File: ios/AnnotationAccPackKit/OTAnnotationAccPackKit/AnnotationImportedViewController.swift
//
//  AnnotationImportedViewController.swift
//
//  Copyright © 2016 Tokbox, Inc. All rights reserved.
//

import UIKit

class AnnotationImportedViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        navigationController?.navigationBar.isTranslucent = false
        let topOffset = UIApplication.shared.statusBarFrame.size.height + navigationController!.navigationBar.frame.height
        let heightOfToolbar = CGFloat(50)
        
        guard let image = UIImage(named: "mvc") else {
            return;
        }
        let imageView = UIImageView(image: image)
        imageView.frame = CGRect(x: 0, y: 0, width: image.size.width, height: image.size.height)
        imageView.translatesAutoresizingMaskIntoConstraints = false
        
        let screenShareView = OTAnnotationScrollView()
        screenShareView.frame = CGRect(x: 0, y: 0, width: UIScreen.main.bounds.width, height: UIScreen.main.bounds.height - heightOfToolbar - topOffset)
        screenShareView.scrollView.contentSize = image.size
        
        screenShareView.addContentView(imageView)
        
        screenShareView.initializeToolbarView()
        screenShareView.toolbarView!.frame = CGRect(x: 0, y: UIScreen.main.bounds.height - heightOfToolbar - topOffset, width: UIScreen.main.bounds.width, height: heightOfToolbar)
        
        self.view.addSubview(screenShareView)
        self.view.addSubview(screenShareView.toolbarView!)
    }
}
